docs/execution-providers/DirectML-ExecutionProvider.md

```diff
@@ -98,28 +106,40 @@ If creating the onnxruntime InferenceSession object directly, you must set the a
+The DirectML execution provider works most efficiently when tensor shapes are known at the time a session is created.  Here are a few performance benefits:
 
-In this case, there are three options:
-- Edit the model to replace an input's free dimension (specified through ONNX using "dim_param") with a fixed size (specified through ONNX using "dim_value").
-- Specify values of named dimensions within model inputs when creating the session using the OnnxRuntime *AddFreeDimensionOverrideByName* ABI.
-- Edit the model to ensure that an input's free dimension has a [denotation](https://github.com/onnx/onnx/blob/master/docs/DimensionDenotation.md) (such as "DATA_BATCH," or a custom denotation).  Then when creating the session, specify the dimension size for each denotation.  This can be done using the OnnxRuntime *AddFreeDimensionOverride* ABI.
+1. Constant folding can occur more often, reducing the CPU / GPU copies and stalls during evaluations.
+2. More initialization work occurs when sessions are created rather than during the first evaluation.
+3. Weights may be pre-processed within DirectML, enabling more efficient algorithms.
+4. Graph optimization occurs within DirectML. For example, Concat operators may be removed and more optimal tensor layouts may be used for the input and output of operators.
 
+Normally, when the shapes of model inputs are known during a session creation, the shapes for the rest of the model are inferred by OnnxRuntime when that session is created.  
+
+However, if a model input contains a free dimension (such as for batch size), additional steps must be taken to retain the above performance benefits. These include:
+
+1. Edit the model to replace an input's free dimension (specified through ONNX using "dim_param") with a fixed size (specified through ONNX using "dim_value").
+2. Specify values of named dimensions within model inputs when creating the session using the OnnxRuntime *AddFreeDimensionOverrideByName* ABI.
+3. Edit the model to ensure that an input's free dimension has a [denotation](https://github.com/onnx/onnx/blob/master/docs/DimensionDenotation.md) (such as "DATA_BATCH," or a custom denotation).  Then, when creating the session, specify the dimension size for each denotation. This can be done using the OnnxRuntime *AddFreeDimensionOverride* ABI.
+
+<p><a href="#">Back to top</a></p>
 
 ## Samples
 
-A complete sample of onnxruntime using the DirectML execution provider can be found under [samples/c_cxx/fns_candy_style_transfer](https://github.com/microsoft/onnxruntime/tree/master/samples//c_cxx/fns_candy_style_transfer).
+[The DirectML GitHub repo](https://github.com/microsoft/DirectML/blob/master/README.md#directml-samples) has sample models which can be executed by the DirectML Execution Provider.
```

Comment:
Is this right? The new link to DirectML samples are samples that illustrate how to call DML API more *directly*, rather than going through ORT. Wouldn't we want to continue to point to sample showing ORT usage with the DML backend?

Reply:
I agree. Will revert to the original